core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -466,14 +410,77 @@ impl TransactionViewReceiveAndBuffer {
             return Err(PacketHandlingError::ComputeBudget);
         };
 
-        let max_age = calculate_max_age(sanitized_epoch, deactivation_slot, alt_resolved_slot);
+        let max_age = calculate_max_age(root_bank.epoch(), deactivation_slot, root_bank.slot());
         let fee_budget_limits = FeeBudgetLimits::from(compute_budget_limits);
         let (priority, cost) = calculate_priority_and_cost(&view, &fee_budget_limits, working_bank);
 
         Ok(TransactionState::new(view, max_age, priority, cost))
     }
 }
 
+/// Perform sanitization checks and transition from data to an executable
+/// [`RuntimeTransaction`]. This additionally returns the minimum slot for
+/// ALT deactivation, if any. If no minimum slot, Slot::MAX is returned.
+pub(crate) fn translate_to_runtime_view<D: TransactionData>(
+    data: D,
+    bank: &Bank,
+) -> Result<(RuntimeTransaction<ResolvedTransactionView<D>>, u64), PacketHandlingError> {
+    // Parsing and basic sanitization checks
+    let Ok(view) = SanitizedTransactionView::try_new_sanitized(
+        data,
+        bank.feature_set
+            .is_active(&agave_feature_set::static_instruction_limit::id()),
+    ) else {
+        return Err(PacketHandlingError::Sanitization);
+    };
```

Comment:
Uses root bank to check feature activiations, is this okay? it previously used working bank

Reply:
I can back this out actually, just pass 2 bank &; for external those refs will just be the same (idk why i thought this would be an issue when writing initially!)